src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1223,7 +1223,7 @@ private static bool FallBackOnDiscard(IdentifierNameSyntax node, DiagnosticBag d
             bool isDiscard = containingDeconstruction != null || IsOutVarDiscardIdentifier(node);
             if (isDiscard)
             {
-                CheckFeatureAvailability(node.Location, MessageID.IDS_FeatureTuples, diagnostics);
+                CheckFeatureAvailability(node, null, MessageID.IDS_FeatureTuples, diagnostics);
```

Comment:
Consider naming `locationOpt` parameter. Maybe move it to last position to avoid naming other parameters too?